src/common/images/editFileDialog.component.tsx

```diff
@@ -0,0 +1,196 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+  Grid,
+  TextField,
+} from '@mui/material';
+import { useForm } from 'react-hook-form';
+
+import React from 'react';
+
+import { UseMutationResult } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { APIImage, ObjectFilePatch } from '../../api/api.types';
+import { FileSchema } from '../../form.schemas';
+import handleIMS_APIError from '../../handleIMS_APIError';
+
+export interface FileDialogProps {
+  open: boolean;
+  onClose: () => void;
+  selectedFile?: APIImage;
+  fileType: 'Image' | 'Attachment';
+  usePatchFile: () => UseMutationResult<
+    APIImage,
+    AxiosError,
+    { id: string; fileMetadata: ObjectFilePatch }
+  >;
+}
```

Comment:
Can use use a discrimination union for the type. to distinguish between image and attachments type  
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html